specs/experimental/zk-fault-dipute-game/overview.md

```diff
@@ -23,4 +37,91 @@
+
+#### Guest Program
+
+In the case of the ZK fault proof, the guest program is an extension of [L2 Chain Derivation][g-l2-chain-derivation]
+that includes a connectivity check among the blocks from L1 origin block and the specified block `C`. The hash of
+Block `C` is determined as the parent hash stored when the [Dispute Game][g-dispute-game] is created in Dispute Game
+Factory by calling `create()` of the Dispute Game Factory.
+If the attacker manipulates any data within the extension of L2 Chain Derivation, it will affect the block hash `C`.
+This is because all data is ultimately linked to the block hash `C` through the hash chain. Therefore if any data
+manipulating attack be thwarted by checking the block hash `C` value. For example, if an attacker creates a proof
+using a transaction that is not included in the [sequencer-batch][g-sequencer-batch], the block `C` value will change,
+preventing them from winning the challenge.
+
+#### Host Program
+
+The host program is a main part of the prover, responsible for delivering the guest program and its input to the zkVM.
+The host program first executes the guest program logic to gather the necessary data, which is equal to what is stored
+in the [PreImageOracle] during the [Optimism's Dispute Game], but the block headers from the L1 block containing
+the [sequencer-batch][g-sequencer-batch] up to block `C` must be included. Finally, the host program delivers the guest
+program and the preimages to the zkVM to obtain the zkVM proof.
+
+[PreImageOracle]: https://github.com/ethereum-optimism/specs/blob/main/specs/fault-proof/stage-one/fault-dispute-game.md#preimageoracle
+[Optimism's Dispute Game]: https://github.com/ethereum-optimism/specs/blob/main/specs/fault-proof/stage-one/fault-dispute-game.md#fault-dispute-game
+
+#### Public Values of Proof
+
+To mark which blocks have been executed, the proof publicly reveals the following data.
+
+``` plain
+1. Output Root at parent block of the target
+2. Output Root at the target block
+3. Hash of the block C
+```
+
+#### Zk Verify contract
+
+The proof generated by a prover can be verified on a verifier contract that includes the following interface.
+The verification in a challenge process is implemented using the verify function provided by the verifier contract.
+
+``` solidity
+interface ZKVerifier {
+    /// @notice The entrypoint for verifying the proof of zk fault proof.
+    /// @param _proofBytes The encoded proof.
+    /// @param _publicValues The encoded public values.
+    /// @return parentOutputRoot output root at parent block of the target.
+    /// @return outputRoot output root at the target block.
+    /// @return L1EndHash hash of the l1 end block.
+    function verify(
+      bytes calldata _publicValues, 
+      bytes calldata _proofBytes
+    ) public view returns (
+      bytes32 parentOutputRoot, 
+      bytes32 outputRoot, 
+      bytes32 L1EndHash
+    );
+}
+```
+
+## Prover as an RPC Server
```

Comment:
Do we need to this section for ZKFDG spec? It's more like docs, not spec.

Reply:
Do you mean that we should remove the 'Prover as an RPC Server' section?

Reply:
Yes, what do you think?